core: Undo %pI6 for printing gids

(commit 8867cd7c8678ff2d9d0382dbbfbcc7a3e7e61cbc)
    
Signed-off-by: Jack Morgenstein <jackm@dev.mellanox.co.il>

Index: ofed_kernel-fixes/drivers/infiniband/core/sysfs.c
===================================================================
--- ofed_kernel-fixes.orig/drivers/infiniband/core/sysfs.c	2009-07-20 13:50:08.000000000 +0300
+++ ofed_kernel-fixes/drivers/infiniband/core/sysfs.c	2009-07-20 13:55:05.000000000 +0300
@@ -255,7 +255,15 @@ static ssize_t show_port_gid(struct ib_p
 	if (ret)
 		return ret;
 
-	return sprintf(buf, "%pI6\n", gid.raw);
+	return sprintf(buf, "%04x:%04x:%04x:%04x:%04x:%04x:%04x:%04x\n",
+		       be16_to_cpu(((__be16 *) gid.raw)[0]),
+		       be16_to_cpu(((__be16 *) gid.raw)[1]),
+		       be16_to_cpu(((__be16 *) gid.raw)[2]),
+		       be16_to_cpu(((__be16 *) gid.raw)[3]),
+		       be16_to_cpu(((__be16 *) gid.raw)[4]),
+		       be16_to_cpu(((__be16 *) gid.raw)[5]),
+		       be16_to_cpu(((__be16 *) gid.raw)[6]),
+		       be16_to_cpu(((__be16 *) gid.raw)[7]));
 }
 
 static ssize_t show_port_pkey(struct ib_port *p, struct port_attribute *attr,
